### PR TITLE
AlignDoubleArrowFixer - issue

### PR DESCRIPTION
**Pull request by Soullivaneuh**
@keradus @SpacePossum I'm sorry to say that #1450 is not completely resolved.

Here another proof. That code was not updated before 1.10.1/2.

### CODE REVIEW

File: Symfony/CS/Tests/Fixer/Contrib/AlignDoubleArrowFixerTest.php [hunk 1]
```diff
@@ -448,6 +448,12 @@ public function provideFixCases()
                 ;
                 ',
             ),
+            array(
+                '<?php
+                $dummy001 = $this->get(\'doctrine\')->getRepository(\'AppBundle:Entity\')->findBy([\'server\' => $object], [\'addedAt\' => \'DESC\'], 5);
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/AlignDoubleArrowFixerTest.php]: simply use `"` instead of escape over and over again...

Comment (Soullivaneuh) [inline Symfony/CS/Tests/Fixer/Contrib/AlignDoubleArrowFixerTest.php]: Done.

**Comment by keradus** [comment]
Bug you referred to was introduce in #1422, which was merged into v1.10.1 and fixed in v1.10.2.
Bug you have provided now indeed exists, bug it was already on v1.10.

Please, could you work on a fix?

**Comment by Soullivaneuh** [comment]
> bug it was already on v1.10

I'm sure it wasn't. This piece of code was on my project for a long time and I didn't get this issue on `v1.10`.

**Comment by Soullivaneuh** [comment]
I was wrong, bug exists on `v1.10` too, that's weird.

> Please, could you work on a fix?

Not sure how I could do that. I think we have to have a flag to see if we are an a new separated line or not but I think this is not possible without huge code modification. I'm maybe wrong ofc.

**Comment by keradus** [comment]
> I'm sure it wasn't. This piece of code was on my project for a long time and I didn't get this issue on `v1.10`.

I have checked out on v1.10 tag and pick your commit - test failed.

**Comment by Soullivaneuh** [comment]
@keradus @SpacePossum I found a strange behavior. If I test with:

```php
$dummy001 = $this->get("doctrine")->getRepository("AppBundle:Entity")->findBy(array("server" => $object), array("addedAt" => "DESC"), 5);
$foobar = $this->getDoctrine()->getRepository("AppBundle:Entity")->findBy(array("server" => $object), array("checkedAt" => "desc"), 50);
```

It passes through the short array condition: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/e8b3c4e41dc1484210fdc45363c41af6c2d56f20/Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php#L105

It should not, is it?

**Comment by Soullivaneuh** [comment]
Nerverbind, it passes by long condition. But only once, any idea?

**Comment by Soullivaneuh** [comment]
Ha! I think I found how to fix it! Will update the PR soon.

**Comment by Soullivaneuh** [comment]
@keradus Should be fixed now. :+1:

File: Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php [hunk 2]
```diff
@@ -139,7 +139,7 @@ private function injectAlignmentPlaceholders(Tokens $tokens, $startAt, $endAt)
                 continue;
             }
 
-            if ($token->equals(',')) {
+            if ($token->equals(',') && !in_array($tokens[$index - 1]->getContent(), array(')', ']'), true)) {
```
Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php]: what if we just have array access going on, and no actual arrays?

Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php]: like, say `$foo[123],`

Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php]: could you add that test perhapse?

Comment (Soullivaneuh) [inline Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php]: In this case you'll match `$token->isGivenKind(T_DOUBLE_ARROW)` part and the method will be called again.

Which test exactly you want me to add?

**Comment by Soullivaneuh** [comment]
ping @keradus

**Comment by keradus** [comment]
pong @Soullivaneuh

File: Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php (hunk 2 above)
Comment (keradus) [inline Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php]: need a tests for all `$foo[123]` vs `[1, 2, 3]` and `(1 + 2)` vs `array(1, 2)`

Comment (Soullivaneuh) [inline Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php]: Sorry I don't understand what you want to test exactly.

Please provide complete test data if you want me to add.

Comment (Soullivaneuh) [inline Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php]: @keradus @GrahamCampbell tests added. Hope this is what you want. Tell me what to do if not. :+1:

File: Symfony/CS/Tests/Fixer/Contrib/AlignDoubleArrowFixerTest.php [hunk 3]
```diff
@@ -448,6 +448,42 @@ public function provideFixCases()
...
+                '<?php
+                $dummy001 = $this->getStrangeDummyManager()->findBy((1 + 2));
+                $foobar = $this->get("manager.dummy")->findBy((1 + 2));
+                ',
+            ),
+            array(
+                '<?php
+                $dummy001 = $this->getStrangeDummyManager()->findBy(array(1, 2));
+                $foobar = $this->get("manager.dummy")->findBy(array(1, 2));
+                ',
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/AlignDoubleArrowFixerTest.php]: all tests you have added have different indentation than current tests. Please have them all be consistence

File: Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php (hunk 2 above)
Comment (keradus) [inline Symfony/CS/Fixer/Contrib/AlignDoubleArrowFixer.php]: good for me, @GrahamCampbell have you got anything else on your mind ?

**Comment by Soullivaneuh** [comment]
@keradus done.

**Comment by keradus** [comment]
```php
            array(
                '<?php
    $inflect_male = array(
        "aitė\b" => "as",
    );',
                '<?php
    $inflect_male = array(
        "aitė\b" => "as",
    );',
            ),
```

vs

```php
            array(
                '<?php
                $dummy001 = $this->get("doctrine")->getRepository("AppBundle:Entity")->findBy(array(1, 2));
                $foobar = $this->getDoctrine()->getRepository("AppBundle:Entity")->findBy(array(1, 2));
                ',
            ),
```

see the indent difference? :)

**Comment by Soullivaneuh** [comment]
I see a difference yes, but not the goal.

Do you want me to add the test? Seems already here.

To be more simple and quicker, please just give the test suites you want to add. ;-)

**Comment by keradus** [comment]
No no, I'm only talking about convention here, like writing:
```php
            array(
                '<?php
    $dummy001 = $this->get("doctrine")->getRepository("AppBundle:Entity")->findBy(array(1, 2));
    $foobar = $this->getDoctrine()->getRepository("AppBundle:Entity")->findBy(array(1, 2));
',
            ),
```

instead of
```php
            array(
                '<?php
                $dummy001 = $this->get("doctrine")->getRepository("AppBundle:Entity")->findBy(array(1, 2));
                $foobar = $this->getDoctrine()->getRepository("AppBundle:Entity")->findBy(array(1, 2));
                ',
            ),
```


it's not crucial, but I like the code to look same way

**Pull request closed without merging**